freqtrade/persistence/trade_model.py

```diff
@@ -1118,6 +1128,10 @@ def delete(self) -> None:
         Trade.query.session.delete(self)
         Trade.commit()
 
+        for kval in self.keyvalues:
+            KeyValue.query.session.delete(kval)
+        KeyValue.query.session.commit()
```

Comment:
the sequence of this must be reversed - deleting the trade first will fail as long as a keyvalue is still available as there is a referential key - or the "trade" deletion will automatically delete all kvalues through the foreign key with "cascade all" (i'm not sure if that properly works on sqlite ...).


If it's not for you - make sure both tables migrate again (or use a new database) - as then you'll have the foreign key missing. 

Sqlalchemy will not migrate your tables automatically if they already exist - even if they have a different schema.

Reply:
Fair point. This was not directly tested so not sure how it will behave. Will follow up when I have more info on it.

Reply:
Addressed through https://github.com/freqtrade/freqtrade/pull/6908/commits/c420304b33b94f67533864345201893ea28f2b9b .